SConstruct

```diff
@@ -1229,9 +1229,22 @@ env['python_cmd_esc'] = quoted(env['python_cmd'])
 
 # Python Package Settings
 python_min_version = LooseVersion('3.5')
+# The string is used to set python_requires in setup.py.in
+env['py_min_ver_str'] = str(python_min_version)
 # Note: cython_min_version is redefined below if the Python version is 3.8 or higher
 cython_min_version = LooseVersion('0.23')
-numpy_min_test_version = LooseVersion('1.8.1')
+numpy_min_version = LooseVersion('1.12.0')
+
+# We choose ruamel.yaml 0.15.34 as the minimum version
+# since it is the highest version available in the Ubuntu
+# 18.04 repositories and seems to work. Older versions such as
+# 0.13.14 on CentOS7 and 0.10.23 on Ubuntu 16.04 raise an exception
+# that they are missing the RoundTripRepresenter
+ruamel_min_version = LooseVersion('0.15.34')
+check_for_ruamel_yaml = any(
+    target in COMMAND_LINE_TARGETS
+    for target in ["install", "test", "test-python-convert"]
+)
```

Comment:
I think the explanation for why we make this more complex and don't always check the `ruamel.yaml` version needs to be documented here.